server/planning/types/planning_featured.py

```diff
@@ -0,0 +1,22 @@
+from datetime import datetime
+from pydantic import Field
+from typing import Annotated
+
+from .base import BasePlanningModel
+from superdesk.utc import utcnow
+from superdesk.core.resources import fields
+from superdesk.core.resources.validators import validate_data_relation_async
+
+
+class PlanningFeaturedResourceModel(BasePlanningModel):
+    date: datetime = Field(default_factory=utcnow)
+    items: list = Field(default_factory=list)
+    tz: str
+    posted: bool
+    last_posted_time: datetime = Field(default_factory=utcnow)
+    last_posted_by: Annotated[fields.ObjectId, validate_data_relation_async("users")]
+    firstcreated: datetime = Field(default_factory=utcnow)
+    versioncreated: datetime = Field(default_factory=utcnow)
+    item_type: str = Field(
```

Comment:
This won't be needed, as it's provided in base ResourceModel class, and uses the name using for the resource config instance.

Reply:
@MarkLark86 a question for clarity. 

The base ResourceModel you highlighted class has `created` and `updated` but not `firstcreated` and `versioncreated` which were then updated in the `on_create` by the `update_dates_for` call. For now since the default is utcnow, I've removed the call `update_dates_for` to avoid redundancy cause it was also just setting the current time.

I'm wondering however if I should still retain them cause they might be used elsewhere similar to how the Events and Planning models have both fields. 

Or maybe add them to `BasePlanningModel` together with `original_creator` and `version_creator`. 

Thoughts ?

Reply:
The `firstcreated` and `versioncreated` have a different context to `created` and `updated`.
* `created` & `updated`: Stores the date/time when the item was created/updated in our DB
* `firstcreated` & `versioncreated` may default to utcnow, but can also be used when ingesting items into our system, to indicate when they were initially created (possibly from external system).

Not sure if these fields would be needed by many other models, and probably should just be used ad-hoc when required